src/runtime/proc.go

```diff
@@ -714,6 +714,9 @@ func schedinit() {
 	if n, ok := atoi32(gogetenv("GOMAXPROCS")); ok && n > 0 {
 		procs = n
 	}
+	if sys.GoosIos != 0 {
```

Comment:
move this up to before the gogetenv check so if we do get env vars working, then this doesn't override it?